File: app/src/main/java/com/dovoo/memesnetwork/utils/GlobalFunc.java
package com.dovoo.memesnetwork.utils;

import android.app.Activity;
import android.content.DialogInterface;
import android.support.v7.app.AlertDialog;

import com.dovoo.memesnetwork.testing.DirectLinkItemTest;
import com.google.android.gms.auth.api.signin.GoogleSignInClient;

public class GlobalFunc {

    public static GoogleSignInClient mGoogleSignInClient;
    public static DirectLinkItemTest currentVideoItem;
    public static boolean isMute = false;

    public static void logout(final Activity activity) {
        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
        builder.setMessage("Do you want to logout?")
                .setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        SharedPreferenceUtils.removeUserPrefs(activity.getApplicationContext());
                        activity.finish();
                    }
                })
                .setNegativeButton("No", new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {

                    }
                });
        // Create the AlertDialog object and return it
        builder.create();
        builder.show();
    }
}

File: app/src/main/java/com/dovoo/memesnetwork/utils/GlobalFunc.kt
package com.dovoo.memesnetwork.utils

import android.content.Context
import com.dovoo.memesnetwork.adapter.items.DirectLinkItemTest
import com.google.android.gms.auth.api.signin.GoogleSignIn

object GlobalFunc {

    @JvmField
    var currentVideoItem: DirectLinkItemTest? = null

    @JvmField
    var isMute = false

    @JvmStatic
    fun isLogin(context: Context): Boolean {
        val localLogin = SharedPreferenceUtils.getPrefs(context).getBoolean(
            SharedPreferenceUtils.PREFERENCES_USER_IS_LOGIN,
            false
        )

        return (GoogleSignIn.getLastSignedInAccount(context) != null) && localLogin
    }

    @JvmStatic
    fun successLogin(context: Context) {
        SharedPreferenceUtils.getPrefsEditor(context).putBoolean(
            SharedPreferenceUtils.PREFERENCES_USER_IS_LOGIN,
            true
        )
    }

    @JvmStatic
    fun setLogout(context: Context) {
        SharedPreferenceUtils.getPrefsEditor(context).putBoolean(
            SharedPreferenceUtils.PREFERENCES_USER_IS_LOGIN,
            false
        )
    }

    fun getLoggedInUserId(context: Context): Int {
        val userId = SharedPreferenceUtils.getPrefs(context)
            .getInt(SharedPreferenceUtils.PREFERENCES_USER_ID, -1)
        return userId
    }

    @JvmStatic
    fun getNotifCount(
        context: Context
    ): Int {
        return SharedPreferenceUtils.getPrefs(context)
            .getInt(SharedPreferenceUtils.PREFERENCES_NOTIF_COUNT, 0)
    }

    @JvmStatic
    fun addNotifCount(
        context: Context
    ) {
        val count = SharedPreferenceUtils.getPrefs(context)
            .getInt(SharedPreferenceUtils.PREFERENCES_NOTIF_COUNT, 0)
        SharedPreferenceUtils.setPrefs(
            context,
            SharedPreferenceUtils.PREFERENCES_NOTIF_COUNT,
            count + 1
        )
    }

    @JvmStatic
    fun minNotifCount(
        context: Context
    ) {
        val count = SharedPreferenceUtils.getPrefs(context)
            .getInt(SharedPreferenceUtils.PREFERENCES_NOTIF_COUNT, 0)
        SharedPreferenceUtils.setPrefs(
            context,
            SharedPreferenceUtils.PREFERENCES_NOTIF_COUNT,
            count - 1
        )
    }

    @JvmStatic
    fun clearNotifCount(
        context: Context
    ) {
        SharedPreferenceUtils.setPrefs(context, SharedPreferenceUtils.PREFERENCES_NOTIF_COUNT, 0)
    }
}